src/multivariate/composite.jl

```diff
@@ -0,0 +1,203 @@
+# Composite Multivariate distribution
+
+## Generic Composite multivariate distribution class
+
+module CompositeDistributions
+
+if !isdefined(:Distributions) using Distributions end
+
+import Base.length, Base.mean, Base.show
+import Distributions.params
+import Distributions.mean, Distributions.mode, Distributions.var, Distributions.cov
+import Distributions.entropy, Distributions.insupport
+import Distributions._logpdf, Distributions._pdf!
+import Distributions._rand!
+
+export AbstractCompositeContinuousDist, ContinuousMultivariateDistribution
+export GenericCompositeContinuousDist, CompositeDist
+export length, params, set_params!, mean, mode, var, cov, entropy
+export insupport, _logpdf, gradlogpdf, _rand!
+
+
+abstract AbstractCompositeContinuousDist <: ContinuousMultivariateDistribution
+#abstract AbstractCompositeDiscreteDist <: DiscreteMultivariateDistribution  # An idea, but not implemented yet
+
+immutable GenericCompositeContinuousDist <: AbstractCompositeContinuousDist
+    dist::Vector{ContinuousDistribution}
```

Comment:
But the who point is that I want to be able to have different variables come from different distributions.   Wouldn't your suggestion for a vector of parametric types force each member of the vector to be from the same distribution?